sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
Ah, but this truncation is zone independent[0], it always truncates relative to the zero time. We do probably want to make that truncation client time zone aware, but this PR doesn't make the behaviour any worse. I guess the current behaviour seems to be consistent because `Truncate` happens to operate in UTC, but it's just a coincidence. 

[0] http://play.golang.org/p/HAd47eseDg